src/Compilers/VisualBasic/Portable/Scanner/Scanner.vb

```diff
@@ -2087,15 +2097,15 @@ FullWidthRepeat2:
                 result = DirectCast(result.AddError(ErrorFactory.ErrorInfo(ERRID.ERR_Overflow)), SyntaxToken)
             ElseIf UnderscoreInWrongPlace Then
                 result = DirectCast(result.AddError(ErrorFactory.ErrorInfo(ERRID.ERR_Syntax)), SyntaxToken)
-            End If
-
-            If UnderscoreUsed Then
+            ElseIf LeadingUnderscoreUsed Then
```

Comment:
With `ElseIf` rather than `If` we're no longer reporting both overflow and feature-not-available errors (e.g.: compiling `Dim x = &H123_456_789_ABC_DEF_123` with `-langversion:14`). We should report both errors in such cases since the errors are independent.